app/assets/javascripts/rooms/show/components/votes/finish_votation.coffee

```diff
@@ -1,14 +1,68 @@
 class @FinishVotation
+  constructor: (finish_votation_channel)->
+    @finish_votation_channel = finish_votation_channel 
+    FinishVotationChannel.CALLBACK = FinishVotation.finish_votation_callback
+
+  @finish_votation_callback: ->
+    block_all_cards_from_users()
+
+  block_all_cards_from_users = ->
+    $('.finish-label').show()
+    $('.page-header').data('room-status', 'finish')
+    $('.card.selectable').not('.confirmed').not('.result').find('.inner').css('background', 'gray')
+    $('.card.selectable').not('.result').css('border-color', 'gray')
+
   handler_events: ->
-    handle_on_finish_click()
+    handle_on_finish_click.call(@)
 
   handle_on_finish_click = ->
-    $('#finish_votation').on 'click', ->
-      $('.card-label').each ->
-        @.voted = '?' if @.voted == undefined
+    fv = @finish_votation_channel
+    $('#finish_votation').on  'click', ->
+      fv.finish()
+      remove_all_card_result()
+      calculate_result_from_votation()
       App.management_room.flip_cards()
-      $('.card-label').each ->
-        @.voted = undefined
+
+  remove_all_card_result = ->
+    $('.card-result').remove()
+
+  calculate_result_from_votation = ->
+    for num in [0..1]
+      frequencies = {}
+      $(".step-result-0#{num}").each ->
+        unless @.voted == undefined
+          frequencies[@.voted] = if frequencies[@.voted] == undefined 
+            1
+          else 
+            frequencies[@.voted] + 1
+
+      process_result_and_build_cards(frequencies, num)
+
+  process_result_and_build_cards = (frequencies, step_number) ->
+    max_value = Math.max.apply(null, Object.values(frequencies))
+
+    result = []
+    for key, value of frequencies
+      if value >= max_value
+        result.push key
+
+    html = ''
+
+    for card in result
```

Comment:
That loop can be a private function as you did before. A build_result_cards for example.